local-container-endpoints/handlers/credentials_handler.go

```diff
@@ -51,16 +51,48 @@ type CredentialService struct {
 
 // NewCredentialService returns a struct that handles credentials requests
 func NewCredentialService() (*CredentialService, error) {
```

Comment:
This should work, but I'd prefer if the code only create a single session, and used an endpoints resolver:

```
// NewCredentialService returns a struct that handles credentials requests
func NewCredentialService() (*CredentialService, error) {
	iamCustomEndpoint := utils.GetValue("", config.IAMCustomEndpointVar)
	stsCustomEndpoint := utils.GetValue("", config.STSCustomEndpointVar)

	defaultResolver := endpoints.DefaultResolver()
	customResolverFn := func(service, region string, optFns ...func(*endpoints.Options)) (endpoints.ResolvedEndpoint, error) {
		if service == "iam" && iamCustomEndpoint != "" {
			return endpoints.ResolvedEndpoint{
				URL: iamCustomEndpoint,
			}, nil
		} else if service == "sts" && stsCustomEndpoint != "" {
			return endpoints.ResolvedEndpoint{
				URL: stsCustomEndpoint,
			}, nil
		}
		return defaultResolver.EndpointFor(service, region, optFns...)
	}

	sess, err := session.NewSessionWithOptions(session.Options{
		SharedConfigState: session.SharedConfigEnable,
		Config: aws.Config{
			EndpointResolver = endpoints.ResolverFunc(customResolverFn)
		},
	})
	if err != nil {
		return nil, err
	}

	iamClient := iam.New(sess)
	iamClient.Handlers.Build.PushBackNamed(useragent.CustomUserAgentHandler())
	stsClient := sts.New(sess)
	stsClient.Handlers.Build.PushBackNamed(useragent.CustomUserAgentHandler())
	return NewCredentialServiceWithClients(iamClient, stsClient, sess), nil
}
```

Reply:
The custom endpoints resolver is exactly what I was hoping could be done to reduce the unnecessary second session. Thanks for that pointer!

I've incorporated this change in addition to leveraging some constants in the AWS SDK to eliminate the "iam" and "sts" magic strings.